---
title: "Project-Trial"
author: "Arun Kumar"
date: "13 March 2018"
output: github_document
---

```{r setup, include=FALSE}
library(Rsamtools)
library(DESeq)
library(GenomeInfoDb)
library(GenomicAlignments)
library(ggplot2)
library(tidyverse)
library(reshape2)
library(GEOquery)
library(cluster)
library(limma)
library(knitr)
library(pvclust)
library(xtable)
library(RColorBrewer)
library(plyr)
library(pheatmap)
library(Rtsne)
library(ggdendro)
knitr::opts_chunk$set(echo = TRUE)
```


# STAT 540 Group Project - Splice and Dice

# Loading the Bam files

```{r data}
#loading libraries

#James
combined_gene_sf <- file.path("C:/Users/11jpw/Desktop/Files/UBC Masters 2017/2018 Semester 2/STATS540/SF3B1_rna_seq/2017-01-30_yeast/combined.gene.sf.tpm")
#Arun
#combined_gene_sf <- file.path("E:/Projects/Splice and Dice/SF3B1_rna_seq/2017-01-30_yeast/combined.gene.sf.tpm")
genedata <- read.table(combined_gene_sf)
genedata %>%
  as_tibble()
```
 
First, to confirm that the two mutations did not affect the expression of Hsh155, we do a quick check on the database before averaging the triplicates. We will repeat this process later.

#####Hsh155 expression:

```{r hsh155}
colnames(genedata) <- c("Gene_ID", "K335N1", "K335N2", "K335N3", "P369E1", "P369E2", "P369E3", "WT1", "WT2", "WT3")
genedata = genedata[-1,]

 I <- "YMR288W"
 hsh155expression <- genedata %>% filter(Gene_ID %in% I)
 hsh155expression
 hsh155expression$Gene_ID <- "Hsh155"
 hsh155 <- hsh155expression %>%
  as.data.frame() %>% 
  column_to_rownames("Gene_ID") %>%
  t() %>% as.data.frame() %>% 
  rownames_to_column("sample_id") %>% 
  melt(id = "sample_id") %>% 
  as_tibble() %>% 
  dplyr::select(sample_id,
         gene = variable, 
         expression = value)
 hsh155
 
 ggplot(hsh155, aes(x=sample_id, y=expression)) + geom_point(aes(color=sample_id)) + theme(axis.text.x = element_text(angle=90, hjust=1))
 
```


######Let's shift the focus to the expression of intron containing genes (ICG) in all these replicates.

```{r tub1}

 ID <- c("YML085C","YFL039C","YBR111W-A")
 ICGexpression <- genedata %>% filter(Gene_ID %in% ID)
 ICGexpression
 ICGexpression$Gene_ID <- c("Tub1", "Act1", "Sus1")
 ICG <- ICGexpression %>%
  as.data.frame() %>% 
  column_to_rownames("Gene_ID") %>%
  t() %>% as.data.frame() %>% 
  rownames_to_column("sample_id") %>% 
  melt(id = "sample_id") %>% 
  as_tibble() %>% 
  dplyr::select(sample_id,
         gene = variable, 
         expression = value)
 ICG
 
 ggplot(ICG, aes(x=sample_id, y=expression))+geom_point(aes(color=sample_id)) + theme(axis.text.x = element_text(angle=90, hjust=1)) + facet_wrap(~gene)
 
```

###Filtering the intron containing genes

Instead of loading the intron containing genes one by one, I downloaded the Ares database. 

```{r loading introndataset}
intronlist <- file.path("http://intron.ucsc.edu/yeast4.1/yeastbasic.txt")
introndata <- read.table(intronlist)
introndata %>%
  as_tibble()
```

Filtering the Ares database with the RNA-seq data, we now have a dataset of only intron containing genes. We can check for intron retention, and expression using differential gene expression analysis on this set rather than the whole genome. 

```{r filterdatatointrons}
intronrnaseq <- genedata %>%
  filter(Gene_ID %in% introndata$V2)
intronrnaseq %>%
  as_tibble()
```

###Differential gene expression analysis on ICG's

```{r averagingthedata}
g <- genedata
g <- mutate_all(g, function(x) as.numeric((as.character(x))))
g$Gene_ID <- genedata$Gene_ID
g %>%
  as_tibble()

g$K335N <- rowMeans(subset(g, select = c(K335N1,K335N2,K335N3)), na.rm = TRUE)
g$P369E <- rowMeans(subset(g, select = c(P369E1,P369E2,P369E3)), na.rm = TRUE)
g$WT <- rowMeans(subset(g, select = c(WT1,WT2,WT3)), na.rm = TRUE)
meangenedata <- g[-c(2,3,4,5,6,7,8,9,10)]
meangenedata %>%
  as_tibble()
```

Repeating the Hsh155 expression pattern and for 3 ICGs:
```{r hsh155mean}
 I1 <- "YMR288W"
 hsh155expressionmean <- meangenedata %>% filter(Gene_ID %in% I1)
 hsh155expressionmean
 hsh155expressionmean$Gene_ID <- "Hsh155"
 hsh155mean <- hsh155expressionmean %>%
  as.data.frame() %>% 
  column_to_rownames("Gene_ID") %>%
  t() %>% as.data.frame() %>% 
  rownames_to_column("sample_id") %>% 
  melt(id = "sample_id") %>% 
  as_tibble() %>% 
  dplyr::select(sample_id,
         gene = variable, 
         expression = value)
 hsh155mean
 
 ggplot(hsh155mean, aes(x=sample_id, y=expression)) + geom_point(aes(color=sample_id)) + theme(axis.text.x = element_text(angle=90, hjust=1)) 
```

```{r plottingmeangenedata}

intronrnaseqmean <- meangenedata %>%
  filter(Gene_ID %in% introndata$V2)
intronrnaseqmean %>%
  as_tibble()
```
```{r repeatingexpressioncheckwithmean}
IDD <- c("YML085C","YFL039C","YBR111W-A")
 ICGexpressionmean <- intronrnaseqmean %>% filter(Gene_ID %in% IDD)
 ICGexpressionmean
 ICGexpressionmean$Gene_ID <- c("Tub1", "Act1", "Sus1")
 ICGmean <- ICGexpressionmean %>%
  as.data.frame() %>% 
  column_to_rownames("Gene_ID") %>%
  t() %>% as.data.frame() %>% 
  rownames_to_column("sample_id") %>% 
  melt(id = "sample_id") %>% 
  as_tibble() %>% 
  dplyr::select(sample_id,
         gene = variable, 
         expression = value)
 ICGmean
 
 ggplot(ICGmean, aes(x=sample_id, y=expression)) + geom_point(aes(color=sample_id)) + theme(axis.text.x = element_text(angle=90, hjust=1)) + facet_wrap(~gene)
 
```



```{r meltIRM}
meltintron <- melt(intronrnaseqmean, id="Gene_ID")
meltintron %>%
  as_tibble()

ggplot(meltintron, aes(x=Gene_ID, y=value, color = variable)) + geom_point() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

```{r loggedplotmeltintron}
logmeltintron <- meltintron
logmeltintron$value <- log(meltintron$value,2)
ggplot(logmeltintron, aes(x=Gene_ID, y=value, color = variable)) + geom_point()
```

```{r heatmap}
clust_dist_col = "euclidean"
clust_method = "ward.D2"
clust_scale = "none"  

matr <- data.frame(meangenedata[,-1])
n <- log(matr,2)
n <- subset(n, K335N!=-Inf)
n <- subset(n, P369E!=-Inf)
n <- subset(n, WT!=-Inf)
j <- t(n)
pheatmap(n, na.rm = TRUE, cluster_rows = TRUE, scale=clust_scale, clustering_method = clust_method, clustering_distance_cols = clust_dist_col, show_colnames = T, show_rownames = FALSE, main ="clustering heatmap")

clust_scale = "row"
pheatmap(n, na.rm = TRUE, cluster_rows = TRUE, scale=clust_scale, clustering_method = clust_method, clustering_distance_cols = clust_dist_col, show_colnames = T, show_rownames = FALSE, main ="clustering heatmap")

clust_method = "average"
pheatmap(n, na.rm = TRUE, cluster_rows = TRUE, scale=clust_scale, clustering_method = clust_method, clustering_distance_cols = clust_dist_col, show_colnames = T, show_rownames = FALSE, main ="clustering heatmap")
```

```{r heatmapwithintrondata}
clust_dist_col = "euclidean"
clust_method = "ward.D2"
clust_scale = "none"  

matri <- data.frame(intronrnaseqmean[,-1])
ni <- log(matri,2)
ni <- subset(ni, K335N!=-Inf)
ni <- subset(ni, P369E!=-Inf)
ni <- subset(ni, WT!=-Inf)
ji <- t(ni)
pheatmap(ni, na.rm = TRUE, cluster_rows = TRUE, scale=clust_scale, clustering_method = clust_method, clustering_distance_cols = clust_dist_col, show_colnames = T, show_rownames = FALSE, main ="clustering heatmap")

clust_scale = "row"

pheatmap(ni, na.rm = TRUE, cluster_rows = TRUE, scale=clust_scale, clustering_method = clust_method, clustering_distance_cols = clust_dist_col, show_colnames = T, show_rownames = FALSE, main ="clustering heatmap")

clust_method = "average"

pheatmap(ni, na.rm = TRUE, cluster_rows = TRUE, scale=clust_scale, clustering_method = clust_method, clustering_distance_cols = clust_dist_col, show_colnames = T, show_rownames = FALSE, main ="clustering heatmap")

```